### ISSUE TITLE
Checkpoint 2

### ISSUE THREAD

**Issue by MNuenninghoff** [MNuenninghoff/NPCCreator#3]
@pawaitemadisoncollege Checkpoint 2 is ready for review

**Comment by pawaitemadisoncollege** [comment]
Hi @MNuenninghoff! Great job hitting checkpoint 2 from the DAO/Hibernate standpoint! 

- [x]  Can you add an image of your ERD so that I can review and provide any feedback (if this is here already and I overlooked it, please let me know). If you don't have a preferred way to create the erd, you can generate it out of the database view in IntelliJ. I think the sequence is right click on the db, then diagrams. I realize I should have been more clear about including this for this CP.

- [x] It looks like you implemented the generic dao and added basic unit testing for all entities with regard to crud.  I did not see tests that confirmed all relationships though, for example a User with NPC, and NPC with the various one to many-s. This can help confirm everything is set up correctly before you start implementing the front end. One thing to be sure to verify are delete scenarios - if an NPC is deleted, what should happen with any associations? Test to make sure what should happen, does happen. 

- [x] I noticed the interactionTraits dao - is that still necessary or can it be removed now that you have the generic dao implemented? 

- [ ]  You've been doing a good job updating your time log and project plan as you complete work. Related to the plan, if you haven't reviewed and updated your plan recently for the final 8 weeks, this would be a good time to do that, since you now have all the learning on new topics need for the project (or will have in the next few days). I noticed a few weeks do not have tasks allocated yet.

- [x]  I think I mentioned this in the peer code review issue, but wanted to highlight it here as well since log4j/noprintlns is part of cp 2:https://github.com/search?q=repo%3AMNuenninghoff%2FNPCCreator%20print&type=code

I did a little poking around, mostly to look for best practices, and nothing additional jumped out at me as "missed" at this point. 

I'm going to mark this as "not yet" in blackboard mostly as a reminder to myself that I need to review the ERD yet.

**Comment by MNuenninghoff** [comment]
Hi @pawaitemadisoncollege , I have added a picture of the ERD as it currently stands to the screenshots folder. I also added two additional tests in NOCDaoTest (one checking that when an NPC is deleted, the Ability trait is has is NOT deleted, and one checking that when a user is deleted, their NPCs ARE deleted). I still need to add additional tests confirming the relationships between NPC and the remaining entity classes are functioning properly (e.g. that when an NPC is deleted, all of its various traits do NOT get deleted in the database).
